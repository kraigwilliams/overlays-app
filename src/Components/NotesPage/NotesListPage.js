import React, { Component } from "react";
//import Note from './Note';
import { Link } from "react-router-dom";
import "./notes.css";
import config from "../../config";
import TokenService from "../../Services/token-service";
import { Button } from "../Utils/Utils";


class NoteListPage extends Component {
  
  // when delete button is clicked the topic is deleted

handleDelete = event => {
    event.preventDefault();

    const topicId = this.props.topic.id;
    console.log(topicId, "topicId")
    fetch(`${config.API_ENDPOINT}/topics/${topicId}`, {
      method: "DELETE",
      headers: {
        "content-type": "application/json",
        authorization: "bearer " + TokenService.getAuthToken()
      }
    }).then(this.props.history.push("/"));
  };

  render() {
    //const topicId= this.props.match.params.TopicId
    
    //console.log(slug, "slug")
    return (
      <div className="all-topics">
        <Link to={`/topics/${this.props.topic.slug}`}>
          <h2 className="topics-heading">{this.props.topic.topic_name} </h2>
        </Link>
        <p className="list-contents">
          <a className="visit-url" href={this.props.topic.topic_url}>
            {this.props.topic.topic_url}
          </a>
"
          <Link to={`/topics/${this.props.topic.topic_name}`} >View Notes</Link>

          <Link to="/new-note">
            <i className="fas fa-plus-square"></i> Add Note to Topic
            
          </Link>

          <Button
            className="delete"
            onClick={event => this.handleDelete(event)}
          >
            Delete
          </Button>
        </p>
      </div>
    );
  }
}

export default NoteListPage;
