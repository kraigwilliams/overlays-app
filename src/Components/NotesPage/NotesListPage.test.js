import React from 'react';
//import { render } from '@testing-library/react';
import NotesListPage from './NotesListPage'
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from  'react-dom';
import renderer from 'react-test-renderer';

describe('<App/>',()=>{
  it ('is able to render',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><NotesListPage/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><NotesListPage /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  
})