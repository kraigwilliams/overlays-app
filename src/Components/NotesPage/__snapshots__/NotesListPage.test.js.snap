// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders the UI as expected 1`] = `
<div
  className="all-topics"
>
  <a
    href="/topics/undefined"
    onClick={[Function]}
  >
    <h2
      className="topics-heading"
    >
       
    </h2>
  </a>
  <p
    className="list-contents"
  >
    <a
      className="visit-url"
    />
    <a
      href="/topics/undefined"
      onClick={[Function]}
    >
      View Notes
    </a>
    <a
      href="/new-note"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus-square"
      />
       Add Note to Topic
    </a>
    <button
      className="Button delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </p>
</div>
`;
